### PR TITLE
[dagster-airbyte] Add config to avoid pulling/forwarding Airbyte logs to compute log

### PR DESCRIPTION
**Pull request by benpankow**
### Summary & Motivation

Currently, we poll the [`/jobs/get`](https://airbyte-public-api-docs.s3.us-east-2.amazonaws.com/rapidoc-api-docs.html#post-/v1/jobs/get) API endpoint to get the status of our Airbyte sync. Unfortunately, this endpoint pulls all the Airbyte logs, which can mean the API takes several seconds to return for long (hours) running syncs.

This PR introduces an optional config for the Airbyte resource that disables fetching logs. Under the hood, this instead hits the [`/jobs/list`](https://airbyte-public-api-docs.s3.us-east-2.amazonaws.com/rapidoc-api-docs.html#post-/v1/jobs/list) API endpoint, which lists recent syncs but provides us the same info minus logs. This will hopefully be faster for users hitting this limitation (though it's a bit hacky).

### How I Tested These Changes

Added params to existing unit tests, tested locally with both config options.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9649** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9649" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/9649?utm_source=stack-comment).

**Review by OwenKephart — APPROVED** [review]
nice!

**Comment by benpankow** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/9649?utm_source=stack-comment).


Job Reference: ykk4kBl6Nl5kRQqvJjms

**Pull request merged**
